appformer-ala-wildfly-provider/src/main/java/org/kie/appformer/ala/wildfly/executor/AppFormerWildflyRuntimeExecExecutor.java

```diff
@@ -0,0 +1,230 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.appformer.ala.wildfly.executor;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.UUID;
+import javax.inject.Inject;
+
+import org.guvnor.ala.config.Config;
+import org.guvnor.ala.exceptions.ProvisioningException;
+import org.guvnor.ala.registry.RuntimeRegistry;
+import org.guvnor.ala.wildfly.access.WildflyAccessInterface;
+import org.guvnor.ala.wildfly.access.WildflyAppState;
+import org.guvnor.ala.wildfly.config.WildflyRuntimeConfiguration;
+import org.guvnor.ala.wildfly.executor.WildflyRuntimeExecExecutor;
+import org.guvnor.ala.wildfly.model.WildflyProvider;
+import org.guvnor.ala.wildfly.model.WildflyRuntime;
+import org.kie.appformer.ala.wildfly.config.AppFormerWildflyRuntimeExecConfig;
+import org.kie.workbench.common.screens.datasource.management.backend.integration.wildfly.WildflyBaseClient;
+import org.kie.workbench.common.screens.datasource.management.backend.integration.wildfly.WildflyDataSourceDef;
+import org.kie.workbench.common.screens.datasource.management.model.DataSourceDef;
+import org.kie.workbench.common.screens.datasource.management.model.DataSourceDeploymentInfo;
+import org.kie.workbench.common.screens.datasource.management.model.DriverDef;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AppFormerWildflyRuntimeExecExecutor
+        extends WildflyRuntimeExecExecutor< AppFormerWildflyRuntimeExecConfig > {
+
+    private static final Logger logger = LoggerFactory.getLogger( AppFormerWildflyRuntimeExecExecutor.class );
+
+    private RuntimeRegistry runtimeRegistry;
+
+    private WildflyAccessInterface wildfly;
+
+    private AppFormerProvisioningHelper provisioningHelper;
+
+    @Inject
+    public AppFormerWildflyRuntimeExecExecutor( final RuntimeRegistry runtimeRegistry,
+                                                final WildflyAccessInterface wildfly,
+                                                final AppFormerProvisioningHelper provisioningHelper ) {
+        super( runtimeRegistry, wildfly );
+        this.runtimeRegistry = runtimeRegistry;
+        this.wildfly = wildfly;
+        this.provisioningHelper = provisioningHelper;
+    }
+
+    @Override
+    public Optional< WildflyRuntime > apply( AppFormerWildflyRuntimeExecConfig runtimeConfig ) {
+
+        logger.info( "Starting executor" );
+        final Optional< WildflyProvider > optional = runtimeRegistry.getProvider( runtimeConfig.getProviderId( ), WildflyProvider.class );
+        WildflyProvider wildflyProvider = optional.get( );
```

Comment:
according to the pipeline definition the provider should be there, so I believe it's ok for this stage